### PR TITLE
Feat: Number Formatter

### PR DESCRIPTION
**Pull request by saxenapeeyush**

### CODE REVIEW

File: src/number/index.ts [hunk 1]
```diff
@@ -178,7 +182,7 @@ export function ordinalSuffixOfNumber(num: number): string {
  * millionWithCommas(1000001.12432432); // 1,000,001.12432432
  * ```
  */
-export function millionWithCommas(num: number) {
+export function millionWithCommas(num: number | string) {
```
Comment (harshitpruthi) [inline src/number/index.ts:185]: @param {number | string} add here also

File: src/number/NumberFormatter/index.ts [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+ *    <div style="margin-left:4px">false</div>
+ * </div>
+ * };
+</div>
+ *
+ * @example
+ * ```
+ * NumberFormatter(100023); // '1,00,023.00'
+ * NumberFormatter(100023,PRICE_CURRENCY);  // '₹1,00,023.00'
+ * NumberFormatter(100023,PRICE_CURRENCY_TO_FIXED_ZERO); // '₹1,00,023'
```
Comment (vikaz596) [inline src/number/NumberFormatter/index.ts:150]: toFixed will round off also. That is ... 1.49.toFixed(1) = 1.5

Do we have something that does not round off .... 1.49.toFixedNoRound(1) = 1.4

???

File: src/index.ts [hunk 3]
```diff
@@ -4,3 +4,4 @@ export * from './general';
 export * from './jsx';
 export * from './string';
 export * from './number';
+export * from './number/NumberFormatter/numberConfig';
```
Comment (vikaz596) [inline src/index.ts:7]: Can you comment here that ...
Ideally, numberConfig should be exported from NumberFormatter file but it generates the doc which we don't want and that is why it is exported from here.

File: src/number/NumberFormatter/index.ts (hunk 2 above)
Comment (harshitpruthi) [reply src/number/NumberFormatter/index.ts:150]: toFixedWithoutRounding we have in this function @saxenapeeyush you can use this

File: src/number/NumberFormatter/index.ts [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+  } catch (err) {
+    console.warn(err);
+
+    return currencyAmount;
+  }
+}
+
+
+/**
+ *
```
Comment (dheerajsingh123) [inline src/number/NumberFormatter/index.ts:336]: please add comments

File: src/number/index.ts [hunk 5]
```diff
@@ -334,6 +338,40 @@ export function convertToBillionTrillionFormat(num: string | number, toFixedDeci
...
+ * ```
+ * getNumberSign(1000); // '+'
+ * convertToBillionTrillionFormat('1200'); // '+'
+ * convertToBillionTrillionFormat(-12430); // '-'
+ * convertToBillionTrillionFormat(0); // ''
+ * ```
+ *
+ */
+export function getNumberSign(num: string | number) {
+  const toNumber = Number(num);
```
Comment (dheerajsingh123) [inline src/number/index.ts:360]: put in a try catch block

File: src/number/NumberFormatter/index.ts [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+ *
+ * @param {string} sign
+ * @param {(number | string)} value
+ * @param {boolean} spaceBetweenSignValue
+ * @return {string | number}
+ */
+function returnSignValueStr(sign : string, value : number | string, spaceBetweenSignValue : boolean) {
+  if (isEmpty(sign)) {
```
Comment (dheerajsingh123) [inline src/number/NumberFormatter/index.ts:344]: put in a try catch block

**Review by dheerajsingh123 — APPROVED** [review]

**Review by harshitpruthi — APPROVED** [review]

**Pull request merged**